cpp/tests/copying/slice_tests.cpp

```diff
@@ -131,6 +131,132 @@ TEST_F(SliceStringTest, StringWithNullsAsColumn)
   }
 }
 
+struct SliceListTest : public SliceTest<int> {
+};
+
+TEST_F(SliceListTest, Lists)
+{
+  using LCW = cudf::test::lists_column_wrapper<int>;
+
+  {
+    cudf::test::lists_column_wrapper<int> list{{1, 2, 3},
+                                               {4, 5},
+                                               {6},
+                                               {7, 8},
+                                               {9, 10, 11},
+                                               LCW{},
+                                               LCW{},
+                                               {-1, -2, -3, -4, -5},
+                                               {-10},
+                                               {-100, -200}};
+
+    std::vector<cudf::size_type> indices{1, 3, 2, 4, 1, 9};
+
+    std::vector<cudf::test::lists_column_wrapper<int>> expected;
+    expected.push_back(LCW{{4, 5}, {6}});
+    expected.push_back(LCW{{6}, {7, 8}});
+    expected.push_back(
+      LCW{{4, 5}, {6}, {7, 8}, {9, 10, 11}, LCW{}, LCW{}, {-1, -2, -3, -4, -5}, {-10}});
+
+    std::vector<cudf::column_view> result = cudf::slice(list, indices);
+    EXPECT_EQ(expected.size(), result.size());
+
+    for (unsigned long index = 0; index < result.size(); index++) {
+      cudf::test::expect_columns_equivalent(expected[index], result[index]);
+    }
+  }
+
+  {
+    cudf::test::lists_column_wrapper<int> list{{{1, 2, 3}, {4, 5}},
+                                               {LCW{}, LCW{}, {7, 8}, LCW{}},
+                                               {{{6}}},
+                                               {{7, 8}, {9, 10, 11}, LCW{}},
+                                               {LCW{}, {-1, -2, -3, -4, -5}},
+                                               {LCW{}},
+                                               {{-10}, {-100, -200}}};
+
+    std::vector<cudf::size_type> indices{1, 3, 3, 6};
+
+    std::vector<cudf::test::lists_column_wrapper<int>> expected;
+    expected.push_back(LCW{{LCW{}, LCW{}, {7, 8}, LCW{}}, {{{6}}}});
+    expected.push_back(LCW{{{7, 8}, {9, 10, 11}, LCW{}}, {LCW{}, {-1, -2, -3, -4, -5}}, {LCW{}}});
```

Comment:
initializer list? other cases as well below

Reply:
So I tried this, but there were some compile errors I couldn't get past.  It didn't seem to like the fact that there wasn't a copy constructor for the wrapper.  `fixed_width_column_wrapper` exhibits the same behavior.   It would be non-trivial to add that constructor.  Can we punt on this for now?